src/DynamoCoreWpf/ViewModels/PackageManager/PackageViewModel.cs

```diff
@@ -194,7 +194,16 @@ private void Deprecate()
 
         private bool CanDeprecate()
         {
-            return dynamoViewModel.Model.AuthenticationManager.HasAuthProvider;
+            PackageInfo pkg = new PackageInfo(Model.Name, new Version(Model.VersionName));
+            bool isAuthor = false;
+            var header = packageManagerClient.GetPackageHeader(pkg);
+            if (header != null) {
+                if (header.maintainers.First().username.Equals(dynamoViewModel.Model.AuthenticationManager.Username))
```

Comment:
@mjkkirschner @aparajit-pratap @gregmarr I forgot if maintainers now is mostly a list with only one maintainer? Any case you have remembered with multiple maintainers?

Reply:
there are some packages which have multiple maintainers, I think we should check against all of them with an any statement.

Reply:
`if(headers.maintainers.Any(maintainer=>maintainer.username.Equals(yadayada.Username))`

Reply:
I agree

Reply:
Ok, will update accordingly.

Reply:
I guess there is something a little funny about a packageViewModel knowing about the login state of a specific user  - I guess one thought would be to bubble up a method which returns true or false and does this check to the PackageManagerClient itself?

Ie something like:

```
class packageManagerClient

public boolean DoesCurrentUserOwnPackage(Package package){
 //do stuff here
return result
}